### PR TITLE
Add Examples to six Style/* Cops

### PR DESCRIPTION
**Pull request by garettarrowood**
This PR adds examples to six `Style` Cops listed in the `"Add examples in the documentation"` Issue #4910 .  At the time of writing, none of these Cops have any open "Add example" PRs associated with them.  Each example set is contained to one commit.

Those Cops are:
1. `Style/BarePercentLiterals`
2. `Style/BlockComments`
3. `Style/BlockDelimiters`
4. `Style/ClassCheck`
5. `Style/ColonMethodCall`
6. `Style/Documentation`

-----------------

* [x] Wrote [good commit messages][1].
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] All tests(`rake spec`) are passing.
* [x] The new code doesn't generate RuboCop offenses that are checked by `rake internal_investigation`.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by garettarrowood** [comment]
Looks like AppVeyor runs on Windows and has been having setup issues since Nov 2nd.

All commits since then have this console error:
```
'C:\Ruby22\bin\gem" update --system --no-document' is not recognized as an internal or external command,
operable program or batch file.
```

This is the first master commit that seems to have run into it. 
https://github.com/bbatsov/rubocop/commit/fc9d225e09361684552691a259205e03d2b51cc9 -
 https://ci.appveyor.com/project/bbatsov/rubocop/build/1301/job/i0698677ymj9yq55 

It appears to be a bug with RubyGems 2.7.0, as that commit is the first one to introduce the new minor version (previous commit ran with RubyGems 2.6.14 - https://ci.appveyor.com/project/bbatsov/rubocop/build/1293/job/ylji5he97gtv8lgn )

File: lib/rubocop/cop/style/bare_percent_literals.rb [hunk 1]
```diff
@@ -4,6 +4,14 @@ module RuboCop
   module Cop
     module Style
       # This cop checks if usage of %() or %Q() matches configuration.
+      #
+      # @example
```
Comment (bbatsov) [inline lib/rubocop/cop/style/bare_percent_literals.rb]: Configurable cops should have documentation like the one here https://github.com/bbatsov/rubocop/blob/c329653bae8ceaaa46f90baaa83abf1209f416a4/lib/rubocop/cop/naming/method_name.rb

(good and bad examples depend on what's configured)

File: lib/rubocop/cop/style/bare_percent_literals.rb [hunk 2]
```diff
@@ -4,6 +4,14 @@ module RuboCop
...
+      #   # bad -
```
Comment (bbatsov) [inline lib/rubocop/cop/style/bare_percent_literals.rb]: The `-` here is redundant, so it's best to remove it.

**Comment by garettarrowood** [comment]
WIP - I believe I need to correct the `semantic` configurable examples I have here. I will squash the fix into the last commit.

**Comment by bbatsov** [comment]
Excellent work! Thanks! 👍

**Pull request merged**